File: test/fixtures/benchmarks/java/BadLint.java
// Phase 10k.1.0 — per-language lint fixture, Java row.
//
// Multiple deliberate PMD violations so at least one fires under
// PMD's quickstart ruleset across version drift:
//   - UnusedImports: `import java.util.*;` (wildcard + unused)
//   - EmptyMethodInAbstractClassShouldBeAbstract / EmptyControlStatement:
//     `void unused() {}`
//   - AvoidLiteralsInIfCondition / shorthand-magic numbers in math
//
// The Java pack's lint capability (target: PMD) invokes
// `pmd check -d <fixture> -R rulesets/java/quickstart.xml -f json`
// and reports the resulting JSON `violations[]` count. Asserts
// dxkit's quality pipeline reports `metrics.lintTool === 'pmd'`
// once the lint provider lands in 10k.1.x.

import java.util.*; // PMD: UnusedImports

public class BadLint {
    void unused() {} // PMD: EmptyMethodInAbstractClassShouldBeAbstract / similar

    int magic() {
        return 42 * 2; // PMD: AvoidLiteralsInIfCondition / generic magic-number warns
    }
}

File: test/fixtures/benchmarks/kotlin/BadLint.kt
// Phase 10i.0.2 — per-language lint fixture, Kotlin row.
//
// Multiple deliberate detekt violations so at least one fires under
// detekt's default ruleset across version drift:
//   - WildcardImport: `import com.example.*`
//   - EmptyFunctionBlock: `fun unused() { }`
//   - MagicNumber: literal 42 inline (default-disabled in some
//     versions but harmless if it doesn't fire)
//
// The kotlin pack's lint capability invokes detekt with `--input .`
// and reports the resulting Checkstyle XML's `<error severity=...>`
// counts via `parseDetektCheckstyleXml`. Asserts dxkit's quality
// pipeline reports `metrics.lintTool === 'detekt'` and a non-zero
// (errors + warnings) total.

import java.util.* // detekt: WildcardImport

class BadLint {
    fun unused() { } // detekt: EmptyFunctionBlock

    fun magic(): Int {
        return 42 * 2 // detekt: MagicNumber (when enabled)
    }
}
